### ISSUE TITLE
"Remove redundant assignment" false positive

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#42754]
**Version Used**: 
Version 16.6.0 Preview 3.0 [29923.199.master]

Without this assignment TryGetLastFileWriteTimeUtc will set result to DateTime.FromFileTimeUtc(0) when a file does not exist.

**Steps to Reproduce**:

``` C#
        public bool TryGetLastFileWriteTimeUtc(string path, DateTime? result)
        {
            result = File.GetLastWriteTimeUtc(path);
            if (result != DateTime.FromFileTimeUtc(0))
            {
                return true;
            }
            result = null; // Remove redundant assignment
            return false;
        }
```

**Expected Behavior**:
No messages

**Actual Behavior**:
Message	IDE0059	Unnecessary assignment of a value to 'result'	Microsoft.VisualStudio.ProjectSystem.Managed	c:\project-system\src\Microsoft.VisualStudio.ProjectSystem.Managed\IO\Win32FileSystem.cs	73	Active

**Comment by davkean** [comment]
Lol, this is my bad - it was correct, I didn't the result as `out`
